setup/src/Magento/Setup/Model/ConfigOptionsList/Directories.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Setup\Model\ConfigOptionsList;
+
+use Magento\Framework\App\DeploymentConfig;
+use Magento\Framework\Config\ConfigOptionsListConstants;
+use Magento\Framework\Config\Data\ConfigData;
+use Magento\Framework\Config\File\ConfigFilePool;
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Setup\ConfigOptionsListInterface;
+use Magento\Framework\Setup\Option\SelectConfigOption;
+
+/**
+ * Deployment configuration options for the document root root
+ */
+class Directories implements ConfigOptionsListInterface
+{
+    public const INPUT_KEY_DOCUMENT_ROOT_IS_PUB = 'document-root-is-pub';
+
+    /**
+     * @inheritdoc
+     */
+    public function getOptions()
+    {
+        return [
+            new SelectConfigOption(
+                self::INPUT_KEY_DOCUMENT_ROOT_IS_PUB,
+                SelectConfigOption::FRONTEND_WIZARD_RADIO,
+                ['no', 'yes'],
+                ConfigOptionsListConstants::CONFIG_PATH_DOCUMENT_ROOT_IS_PUB,
+                'Is the webserver\'s document root set to ./pub/',
+                'no'
+            ),
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function createConfig(array $options, DeploymentConfig $deploymentConfig)
+    {
+        $configData = new ConfigData(ConfigFilePool::APP_ENV);
+        if (isset($options[self::INPUT_KEY_DOCUMENT_ROOT_IS_PUB])) {
+            $configData->set(
+                ConfigOptionsListConstants::CONFIG_PATH_DOCUMENT_ROOT_IS_PUB,
+                $this->processBooleanConfigValue($options[self::INPUT_KEY_DOCUMENT_ROOT_IS_PUB])
+            );
+        }
+        return $configData;
+    }
+
+    /**
+     * Provide config value from input.
+     *
+     * @param string $value
+     *
+     * @return bool
+     * @throws InputException
+     */
+    private function processBooleanConfigValue(string $value): bool
+    {
+        if (strtolower($value) === 'true' || $value === '1' || strtolower($value) === 'yes') {
+            return true;
+        }
+
+        if (strtolower($value) === 'false' || $value === '0' || strtolower($value) === 'no') {
+            return false;
+        }
+
+        throw new InputException(__('%1 requires a boolean value.', self::INPUT_KEY_DOCUMENT_ROOT_IS_PUB));
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(array $options, DeploymentConfig $deploymentConfig)
```

Comment:
Until it get merged - can we add there return types?
```suggestion
    public function validate(array $options, DeploymentConfig $deploymentConfig): array
```